cli/azd/extensions/microsoft.azd.extensions/internal/github/github.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/extensions/microsoft.azd.extensions/internal"
+	"github.com/azure/azure-dev/cli/azd/pkg/common"
```

Comment:
Is it expected that extensions can import from `/azd/pkg` ?  (out of `/azd/extensions`) ??   Seems like a shortcut to bypass AZD cli to get AZD's functionality. Something possible when using GO, but maybe something we might want to avoid?

Reply:
Yeah - i was torn about this.  Yes, go extensions can use any packages in `azd` if we want - I was just considering whether this was good or bad practice.  I had created the `azdext` as a start of exposing stuff for extensions.

Happy to just add the new commands if we are OK with this approach.